src/utils/arrow.rs

```diff
@@ -34,6 +36,70 @@ fn get_arrow_column_type(data: &Range<CalData>, row: usize, col: usize) -> Resul
     }
 }
 
+static FLOAT_TYPES_CELL: OnceLock<HashSet<ArrowDataType>> = OnceLock::new();
+static INT_TYPES_CELL: OnceLock<HashSet<ArrowDataType>> = OnceLock::new();
+static STRING_TYPES_CELL: OnceLock<HashSet<ArrowDataType>> = OnceLock::new();
+
+fn float_types() -> &'static HashSet<ArrowDataType> {
+    FLOAT_TYPES_CELL.get_or_init(|| {
+        HashSet::from([
+            ArrowDataType::Int64,
+            ArrowDataType::Float64,
+            ArrowDataType::Boolean,
+        ])
+    })
+}
+
+fn int_types() -> &'static HashSet<ArrowDataType> {
+    INT_TYPES_CELL.get_or_init(|| HashSet::from([ArrowDataType::Int64, ArrowDataType::Boolean]))
+}
+
+fn string_types() -> &'static HashSet<ArrowDataType> {
+    STRING_TYPES_CELL.get_or_init(|| {
+        HashSet::from([
+            ArrowDataType::Int64,
```

Comment:
the case boolean + string is impossible?

Reply:
It is probably possible, but string to bool conversion is not supported by calamine